### PR TITLE
Modify cs.Cfg.BUS creation

### PR DESCRIPTION
**Pull request by BrentHoltsclaw**
Cfg.BUS will keep bus information all pci devices enumerated

Signed-off-by: brentholtsclaw <brent.holtsclaw@intel.com>

### CODE REVIEW

File: chipsec/chipset.py [hunk 1]
```diff
@@ -628,19 +629,44 @@ def init_cfg_bus( self ):
...
+                        elif '-' in tdid:
+                            rdv = tdid.split('-')
+                            for rdv_value in range(int(rdv[0], 16), int(rdv[1], 16) + 1):
+                                did_list.append(rdv_value)
+                        else:
+                            did_list.append(int(tdid, 16))
+                for did in did_list:
+                    cfg_str = "{:0>2}_{:0>2}_{}_{}".format(device_data['dev'][2:] if len(device_data['dev']) > 2 else device_data['dev'],device_data['fun'],device_data['vid'][2:],device_data['did'][2:])
+                    if cfg_str in self.Cfg.BUS.keys():
+                        self.Cfg.BUS[config_device] = self.Cfg.BUS.pop(cfg_str)
```
Comment (iherna2) [inline chipsec/chipset.py:662]: Will this replace an item by itself?

Comment (iherna2) [reply chipsec/chipset.py:662]: This code is not obvious and a comment would help here.

**Review by iherna2 — APPROVED** [review]

File: chipsec/chipset.py [hunk 2]
```diff
@@ -628,19 +628,41 @@ def init_cfg_bus( self ):
...
             device_data = self.Cfg.CONFIG_PCI[config_device]
             xml_vid  = device_data.get( 'vid', None )
             xml_did  = device_data.get( 'did', None )
-            if (xml_vid and xml_did):
-                bus_list = []
-                did_list = [int(_, 16) for _ in xml_did.split(',')]
-                for enum_dev in enum_devices:
-                    if ((int(device_data['dev'], 16), int(device_data['fun'], 16), int(xml_vid, 16)) == enum_dev[1:4]) and (enum_dev[4] in did_list):
-                        bus_list.append( hex(enum_dev[0]) )
-                        if logger().DEBUG: logger().log( '    + {:16s}: VID 0x{:04X} - DID 0x{:04X} -> Bus 0x{:02X}'.format(config_device, enum_dev[3], enum_dev[4], enum_dev[0]) )
```
Comment (frinzell) [inline chipsec/chipset.py:641]: Having this as debug output might be helpful.

File: chipsec/chipset.py [hunk 3]
```diff
@@ -628,19 +628,41 @@ def init_cfg_bus( self ):
...
-                if len(bus_list):
-                    self.Cfg.BUS[ config_device ] = bus_list
+            # if the vid and did are present within the configuration file attempt to replace generic name with configuration name
+            if xml_vid and xml_did:
+                did_list = []
+                # gather list of device id: device id may have single entry, multiple entries, end in "X", or specified by a range "-"
+                if xml_did:
+                    for tdid in xml_did.split(','):
+                        if tdid[-1].upper() == "X":
+                            for rdv_value in range(tdid, tdid+0xF):
```
Comment (frinzell) [inline chipsec/chipset.py:652]: If `tdid[-1]` is an `'x'`, should it be replaced with 0 and converted to an **int**?   And should the `+0xF` be `+0x10`, assuming x is 0?

File: chipsec/chipset.py [hunk 4]
```diff
@@ -628,19 +628,41 @@ def init_cfg_bus( self ):
...
-                for enum_dev in enum_devices:
-                    if ((int(device_data['dev'], 16), int(device_data['fun'], 16), int(xml_vid, 16)) == enum_dev[1:4]) and (enum_dev[4] in did_list):
-                        bus_list.append( hex(enum_dev[0]) )
-                        if logger().DEBUG: logger().log( '    + {:16s}: VID 0x{:04X} - DID 0x{:04X} -> Bus 0x{:02X}'.format(config_device, enum_dev[3], enum_dev[4], enum_dev[0]) )
-                if len(bus_list):
-                    self.Cfg.BUS[ config_device ] = bus_list
+            # if the vid and did are present within the configuration file attempt to replace generic name with configuration name
+            if xml_vid and xml_did:
+                did_list = []
+                # gather list of device id: device id may have single entry, multiple entries, end in "X", or specified by a range "-"
```
[the remark this answers is not in the document]
Comment (BrentHoltsclaw) [reply chipsec/chipset.py:648]: This is the same logic that was used within line 424.

File: chipsec/chipset.py (hunk 2 above)
Comment (BrentHoltsclaw) [reply chipsec/chipset.py:641]: assuming this means you a similar debug statement

File: chipsec/chipset.py (hunk 3 above)
Comment (BrentHoltsclaw) [reply chipsec/chipset.py:652]: Looks like a line is missing, will update.

File: chipsec/chipset.py [hunk 5]
```diff
@@ -628,19 +628,41 @@ def init_cfg_bus( self ):
...
+                                did_list.append(rdv_value)
+                        elif '-' in tdid:
+                            rdv = tdid.split('-')
+                            for rdv_value in range(int(rdv[0], 16), int(rdv[1], 16) + 1):
+                                did_list.append(rdv_value)
+                        else:
+                            did_list.append(int(tdid, 16))
+                # If there is a match between the configuration entry and generic entry, replace the name with the configuration entry
+                for tdid in did_list:
+                    cfg_str = "{:0>2}_{:0>2}_{}_{:04X}".format(device_data['dev'][2:] if len(device_data['dev']) > 2 else device_data['dev'], device_data['fun'], device_data['vid'][2:], tdid)
```
[the remark this answers is not in the document]
Comment (BrentHoltsclaw) [reply chipsec/chipset.py:662]: in 634 the data was converted to integers from pci enumerate. When using the data from the xml, it is strings, so formatting will be a little different.

File: chipsec/chipset.py (hunk 1 above)
Comment (BrentHoltsclaw) [reply chipsec/chipset.py:664]: should be resolved with the break.

File: chipsec/chipset.py [hunk 6]
```diff
@@ -628,19 +628,43 @@ def init_cfg_bus( self ):
...
-                    if ((int(device_data['dev'], 16), int(device_data['fun'], 16), int(xml_vid, 16)) == enum_dev[1:4]) and (enum_dev[4] in did_list):
-                        bus_list.append( hex(enum_dev[0]) )
-                        if logger().DEBUG: logger().log( '    + {:16s}: VID 0x{:04X} - DID 0x{:04X} -> Bus 0x{:02X}'.format(config_device, enum_dev[3], enum_dev[4], enum_dev[0]) )
-                if len(bus_list):
-                    self.Cfg.BUS[ config_device ] = bus_list
+            # if the vid and did are present within the configuration file attempt to replace generic name with configuration name
+            if xml_vid and xml_did:
+                did_list = []
+                # gather list of device id: device id may have single entry, multiple entries, end in "X", or specified by a range "-"
+                if xml_did:
```
Comment (mjkyung) [inline chipsec/chipset.py:649]: Isn't this condition duplicate?

**Review by mjkyung — APPROVED** [review]

File: chipsec/chipset.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (npmitche) [reply chipsec/chipset.py:648]: Could do something like:
``` python
x = re.search(r'((0X)?[0-9A-F]+)(X*)', tdid.upper())
if x[3]:
    tndid = int(x[1], 16) << 4 * len(x[3])
```
... in the future.

**Review by npmitche — APPROVED** [review]

**Review by frinzell — APPROVED** [review]

**Pull request merged**